backend/src/Book/Infrastructure/FileHandler.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace App\Book\Infrastructure;
+
+use Symfony\Component\Serializer\Encoder\CsvEncoder;
+use Symfony\Component\Serializer\Normalizer\ObjectNormalizer;
+use Symfony\Component\Serializer\Serializer;
+
+class FileHandler
+{
+    protected $serializer;
+    private $delimiter = ';';
+
+    public function __construct() 
+        {
+            $this->serializer = new Serializer([new ObjectNormalizer()], [new CsvEncoder()]); 
+        }
+
+    /**
+     * @param mixed $file
+     * @param string $filePath
+     * @param string $delimiter
+     * @return mixed[]
+     */
+    public function csvToArray(mixed $file, string $filePath, string $delimiter = ';'): array
+    {   
+        $this->delimiter = $delimiter;
+        $data = $this->serializer->decode(file_get_contents($filePath), 'csv', [CsvEncoder::DELIMITER_KEY => $delimiter]);
+        $headers = array_keys($data[0]);
+        $dataLen = count($data);
+        return [$headers, $data, $dataLen];
```

Comment:
**nice-to-have:** se non ricordo male avevamo detto di non utilizzare gli array ma gli oggetti giusto 😀? 

Considerate che chi utilizza questo metodo deve conoscere obbligatoriamente la struttura dati di ritorno